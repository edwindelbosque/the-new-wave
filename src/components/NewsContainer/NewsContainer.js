import React from 'react';
import { NewsArticle } from '../../index';
import './NewsContainer.scss';

const NewsContainer = ({ newsArticles }) => {
  return (
    newsArticles.map(article =>
      <NewsArticle
        key={article.id}
        id={article.id}
        headline={article.headline}
        image={article.img}
        description={article.description}
        url={article.url}
      />)
  );
};

export { NewsContainer };