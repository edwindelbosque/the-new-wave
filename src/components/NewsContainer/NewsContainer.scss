main {
  padding: 50px 40px;
  display: grid;
  grid-gap: 50px 50px;
  grid-template-columns: repeat(auto-fit, minmax(350px, 1fr));
}