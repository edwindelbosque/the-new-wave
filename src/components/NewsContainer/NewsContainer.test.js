import React from 'react';
import NewsContainer from './NewsContainer';
import { shallow } from 'enzyme';

describe('NewsContainer', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
      <NewsContainer
        newsArticles={[{ idea1: 'info' }, { idea2: 'info' }]}
      />
    )
    expect(wrapper).toMatchSnapshot();
  })
})