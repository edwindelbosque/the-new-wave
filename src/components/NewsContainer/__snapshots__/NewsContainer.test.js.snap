// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot with all data passed in correctly 1`] = `
Array [
  <NewsArticle />,
  <NewsArticle />,
]
`;
